Tests/SonarScanner.MSBuild.PreProcessor.Test/ProcessedArgsTests.cs

```diff
@@ -33,6 +36,20 @@ public class ProcessedArgsTests
         private ProcessedArgs args;
         private TestLogger logger;
 
+        private static IEnumerable<object[]> DirectoryCreateExceptions
```

Comment:
Testing for all exception types seems a bit too much, considering that the implementation does not consider the exception type.

Reply:
This is straight from the documentation. These are all exceptions the tested API can raise. If we ever want to specialize on certain exceptions, we have all of them under test already.